src/sysctl.cr

```diff
@@ -0,0 +1,87 @@
+lib LibC
+  {% if flag?(:darwin) || flag?(:freebsd) %}
+    fun sysctlbyname(LibC::Char*, Void*, LibC::SizeT*, Void*, LibC::SizeT) : Int32
+  {% end %}
+end
+
+# Provides access to UNIX kernels state using the `sysctl` system call
+module Sysctl
+  extend self
+
+  # Returns the sysctl named *name* as an integer. Will raise `NotImplemented`
+  # if the sysctl interface doesn't exist on your system (Windows) or isn't
+  # supported yet. Will raise the appropriate `Errno` exception in case of
+  # error.
+  #
+  # ```
+  # Sysctl.get_int("hw.ncpu") # => 8
+  # ```
+  #
+  def get_i32(name : String) : Int32
```

Comment:
I've seen API like that in [kemal-session](https://github.com/kemalcr/kemal-session) shard for instance.